lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,200 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+// output_indexing_map   = (d0, d1, d2) -> (d0, d2)
+//
+
+class ConvertAtenEmbeddingBagPaddingIdxOp : public OpConversionPattern<AtenEmbeddingBagPaddingIdxOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenEmbeddingBagPaddingIdxOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+        return failure();
+    Location loc  = op->getLoc();
+    auto context = op->getContext();
+    Value weight  = adaptor.weight();
+    Value indices = adaptor.indices();
+    Value offsets = adaptor.offsets();
+    Value scale_grad_by_freq = adaptor.scale_grad_by_freq();
+    Value mode = op.mode();
+    Value sparse = adaptor.sparse();
+    Value include_last_offset = adaptor.include_last_offset();
+
+    int64_t m;
+    //TODO: handle mean and max modes
+    //sum mode
+    if (!matchPattern(mode, m_TorchConstantInt(&m))){
+     return rewriter.notifyMatchFailure(op, "mode is expected to be an integer value");
+    }  
+    
+    //get the Embedding dimension
+    Value embeddingDim = getDimOp(rewriter, loc, weight, 1);
+
+    auto weightTy = weight.getType().cast<RankedTensorType>();
+    if (weightTy.getRank() != 2)
+      return rewriter.notifyMatchFailure(op, "weight must be rank 2");
+
+    auto indicesTy = indices.getType().cast<RankedTensorType>();
+    if (indicesTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "indices must be a vector");
+
+    auto offsetsTy = offsets.getType().cast<RankedTensorType>();
+    if (offsetsTy.getRank() != 1)
+      return rewriter.notifyMatchFailure(op, "offsets much be a vector");
+
+    Type weightElemTy = weightTy.getElementType();
+
+    if (m == torch_upstream::Reduction::Sum){
+      
+      SmallVector<AffineExpr> indicesExpr;
+      indicesExpr.push_back(mlir::getAffineDimExpr(1, context)); 
+      
+      auto indicesIndexingMap = AffineMap::get(/*dimCount=*/3, /*symbolCount=*/0, indicesExpr, context);
+      
+      SmallVector<AffineExpr> offsetsExpr;
+      offsetsExpr.push_back(mlir::getAffineDimExpr(0, context));
+      
+      auto offsetIndexingMap = AffineMap::get(/*dimCount=*/3, /*symbolCount=*/0, offsetsExpr, context);
+      
+      SmallVector<AffineExpr> outputExpr;
+      outputExpr.push_back(mlir::getAffineDimExpr(0, context));
+      outputExpr.push_back(mlir::getAffineDimExpr(2, context));
+      
+      auto outputIndexingMap = AffineMap::get(/*dimCount=*/3, /*symbolCount=*/0, outputExpr, context);
+      
+      SmallVector<AffineMap, 3> indexingMaps = {
+        indicesIndexingMap,  //input1, indices
+        offsetIndexingMap,   //input2, offsets     
+        outputIndexingMap,   //output 
+      };
+      
+      SmallVector<StringRef> iteratorTypes(3, getParallelIteratorTypeName());
+
+      
+      SmallVector<Value> sizes = getTensorSizes(rewriter, loc, offsets);
+      sizes.push_back(embeddingDim);
```

Comment:
nit: I think it would be a bit clearer to do

```cpp
SmallVector<Value> sizes{ getDimOp(rewriter, loc, offsets, 0), embeddingDim };
```

to make it clear that the size is for a rank-2 tensor